core/src/main/java/greencity/controller/EmailController.java

```diff
@@ -56,10 +58,24 @@ public ResponseEntity<Object> sendReport(@RequestBody SendReportEmailMessage mes
      * @param message - object with all necessary data for sending email
      * @author Taras Kavkalo
      */
+
+    @Operation(summary = "Change the status of a place and notify the author via email")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = HttpStatuses.OK),
+            @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED)
+    })
     @PostMapping("/changePlaceStatus")
-    public ResponseEntity<Object> changePlaceStatus(@RequestBody SendChangePlaceStatusEmailMessage message) {
+    public ResponseEntity<Object> changePlaceStatus(
+            @Valid @RequestBody SendChangePlaceStatusEmailMessage message,
+            BindingResult bindingResult
+    ) {
+        if (bindingResult.hasErrors()) {
+            return ResponseEntity.badRequest().body(bindingResult.getAllErrors());
```

Comment:
Perhaps better to throw BadRequest and give responsibility to ExceptionHandler to build ResponseEntity?

Reply:
Right. If errors are handled by an error handler, it would be better to transfer this responsibility to it.